lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Azure.DataLake.Store;
+using Microsoft.Azure.Management.DataLake.Store;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureDataLake
+{
+    internal sealed class AzureDataLakeFileSystem : IFileSystem
+    {
+        AdlsClient _client;
+
+        [Inject]
+        private AzureDataLakeFileSystem(AzureDataLakeStoreClient client)
+        {
+            _client = client.GetReference();
+        }
+
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            _client.BulkUpload(localFileName, remoteFileUri.ToString());
+        }
+
+        public void CopyToLocal(Uri remoteFileUri, string localName)
+        {
+            _client.BulkDownload(remoteFileUri.ToString(), localName);
+        }
+
+        public Stream Create(Uri fileUri)
+        {
+            return _client.CreateFile(fileUri.ToString(), IfExists.Overwrite);
+        }
+
+        public void CreateDirectory(Uri directoryUri)
+        {
+            _client.CreateDirectory(directoryUri.ToString());
+        }
+
+        public Uri CreateUriForPath(string path)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void Delete(Uri fileUri)
+        {
+            _client.Delete(fileUri.ToString());
```

Comment:
Although the interface has a void function, the ADLSClient returns a Boolean for delete. You need to reconcile this different. I suggest throwing an IOException if the result is false. The same goes for the other ADLSClient boolean functions.